### PR TITLE
use master branch of metadata only for release server versions

### PR DESCRIPTION
**Pull request by kinarashah**
https://github.com/rancher/rancher/issues/23485

### CODE REVIEW

File: pkg/settings/setting.go [hunk 1]
```diff
@@ -38,7 +38,7 @@ var (
 	PeerServices                      = NewSetting("peer-service", os.Getenv("CATTLE_PEER_SERVICE"))
 	RDNSServerBaseURL                 = NewSetting("rdns-base-url", "https://api.lb.rancher.cloud/v1")
 	RkeVersion                        = NewSetting("rke-version", "")
-	RkeMetadataConfig                 = NewSetting("rke-metadata-config", getMetadataConfig())
+	RkeMetadataConfig                 = NewSetting("rke-metadata-config", "")
```
Comment (alena1108) [inline pkg/settings/setting.go]: will it be reset on the upgraded setups?

File: pkg/settings/setting.go [hunk 2]
```diff
@@ -143,19 +143,36 @@ func NewSetting(name, def string) Setting {
...
 		"url":                      "https://github.com/rancher/kontainer-driver-metadata.git",
-		"branch":                   "master",
+		"branch":                   branch,
 		"refresh-interval-minutes": "1440",
 	}
 	ans, err := json.Marshal(data)
-	if err == nil {
-		return string(ans)
+	if err != nil {
+		return ""
```
Comment (alena1108) [inline pkg/settings/setting.go]: do you think we should log an error? So it doesn't get lost

File: pkg/settings/setting.go [hunk 3]
```diff
@@ -143,19 +143,36 @@ func NewSetting(name, def string) Setting {
...
 
-func GetEnvKey(key string) string {
-	return "CATTLE_" + strings.ToUpper(strings.Replace(key, "-", "_", -1))
+func releaseServerVersion(serverVersion string) bool {
+	if serverVersion == "dev" ||
+		serverVersion == "master" ||
+		serverVersion == "" ||
+		strings.HasSuffix(serverVersion, "-head") ||
+		strings.Contains(serverVersion, "-rc") {
+		return false
```
Comment (prachidamle) [inline pkg/settings/setting.go]: do we have any release versions like "alpha" / "beta"?

File: pkg/settings/setting.go (hunk 2 above)
Comment (kinarashah) [reply pkg/settings/setting.go]: updated to log error

File: pkg/settings/setting.go (hunk 3 above)
Comment (kinarashah) [reply pkg/settings/setting.go]: updated logic to calculate server version

**Comment by alena1108** [comment]
LGTM

**Comment by prachidamle** [comment]
LGTM

File: pkg/settings/setting.go (hunk 3 above)
Comment (deniseschannon) [reply pkg/settings/setting.go]: does this account for `v2.4.0-alpha1` as that's what the alpha versions which are used for preview will be tagged.

**Pull request merged**